core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -252,7 +257,11 @@ public LogicalPlan visitRareTopN(RareTopN node, AnalysisContext context) {
     List<Argument> options = node.getNoOfResults();
     Integer noOfResults = (Integer) options.get(0).getValue().getValue();
 
-    return new LogicalRareTopN(child, node.getCommandType(), noOfResults, fields, groupBys);
+    if (node.getCommandType() != RareTopN.CommandType.TOPOFALL) {
+      return new LogicalRareTopN(child, node.getCommandType(), noOfResults, fields, groupBys);
+    } else {
+      return new LogicalRareTopN(child, RareTopN.CommandType.TOP, noOfResults, fields, groupBys);
```

Comment:
I only made minimal changes for the `RareTopN` to make `topOfAll` work, so here it may look ugly. The `RareTopN.CommandType.TOPOFALL` is merely for this class to know whether to `setIndexScanType` as in [line 245](https://github.com/opensearch-project/sql/pull/697/files#diff-999a9252d26453f7423e8847690c4be5e41196a02fa9b80670483ab77ebf02a1R245)